DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi/Queries/Implementations/GitHubQuery.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using GitHubExtension.Security.WebApi.Exceptions;
+using GitHubExtension.Security.WebApi.Extensions.HttpRequestMessage;
+using GitHubExtension.Security.WebApi.Models;
+using GitHubExtension.Security.WebApi.Queries.Constant;
+using GitHubExtension.Security.WebApi.Queries.Interfaces;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace GitHubExtension.Security.WebApi.Queries.Implementations
+{
+    public class GitHubQuery : IGitHubQuery
+    {
+        private readonly HttpClient _httpClient;
+
+        public GitHubQuery()
+        {
+            _httpClient = new HttpClient();
+        }
+
+        public async Task<GitHubUserModel> GetUserAsync(string token)
+        {
+            var requestUri = string.Format(GitHubQueryConstant.GetUser, token);
+            var message = HttpRequestMessageExtension.CreateMessage(HttpMethod.Get, requestUri);
+
+            var response = await _httpClient.SendAsync(message);
+            if (!response.IsSuccessStatusCode)
+                throw new UnsuccessfullGitHubRequestException();
+
+            var dto = JsonConvert.DeserializeObject<GitHubUserModel>(await response.Content.ReadAsStringAsync());
+            dto.Email = dto.Email ?? await GetPrimaryEmailForUser(token);
```

Comment:
It's not possible, because if user've primary email we get empty email.
If we got empty email, we go to GetPrimaryEmailForUser(string token), and after that we've user's email.